.github/workflows/test.yml

```diff
@@ -20,10 +20,17 @@ jobs:
       - uses: actions/setup-node@v4
         with:
           node-version: "18.x"
+
       - uses: actions/checkout@v4
+
       - run: npm i
       - run: npm run lint
 
+      # run formatting and fail if any changes are detected
+      - run: npm run format
```

Comment:
add `--check` and remove the git commands

Reply:
Also: `format` is not run on the entire repo, only on the `src`/`test` folders, but we have files outside those folders. Prettier has an ignore file where we could add the `build` folder (and anything else that we would like to ignore). If we had that config file, we could then run the format on all files that we usually edit.

Reply:
Good idea to use --check 👍 

I'll put up a separate PR for a prettier ignore file and running format on more of the codebase.